packages/shared/lib/clients/sync.client.ts

```diff
@@ -741,6 +723,41 @@ class SyncClient {
             });
         }
     }
+
+    async triggerInitialSync({
+        syncId,
+        jobId,
+        syncName,
+        nangoConnection,
+        debug
+    }: {
+        syncId: string;
+        jobId?: string;
+        syncName: string;
+        activityLogId: number;
+        nangoConnection: NangoConnection;
+        debug?: boolean;
+    }): Promise<boolean> {
+        jobId = jobId || generateWorkflowId({ id: syncId }, syncName, nangoConnection.connection_id);
+        const syncJobId = await createSyncJob(syncId, SyncType.INITIAL, SyncStatus.RUNNING, jobId, nangoConnection);
+
+        if (!syncJobId) {
+            return false;
+        }
+
+        const handle = await this.client?.workflow.start('initialSync', {
+            taskQueue: SYNC_TASK_QUEUE,
+            workflowId: jobId,
+            args: [{ syncId: syncId, syncJobId: syncJobId.id!, nangoConnection, syncName, debug }]
```

Comment:
Is the non null assertion needed here if you check for `syncJobId` earlier?

Reply:
The type `Job` is incorrect it marks id as optional, I guess because the same type is used for creation and for response. 
This is something to fix in a types improvement initiative I think

Reply:
Gotcha, agreed